src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -95,14 +95,15 @@ public void delete(Query query, Class<?> clazz, IndexCoordinates index) {
 
 	// region SearchOperations
 	@Override
-	public <T> CloseableIterator<T> stream(Query query, Class<T> clazz, IndexCoordinates index) {
+	public <T> CloseableIterator<SearchHit<T>> stream(Query query, Class<T> clazz, IndexCoordinates index) {
 		long scrollTimeInMillis = TimeValue.timeValueMinutes(1).millis();
 		return StreamQueries.streamResults(startScroll(scrollTimeInMillis, query, clazz, index),
 				scrollId -> continueScroll(scrollId, scrollTimeInMillis, clazz), this::clearScroll);
 	}
 
 	@Override
-	public <T> Page<T> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index) {
+	public <T> AggregatedPage<SearchHit<T>> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz,
```

Comment:
How does `AggregatedPage` relate to `SearchHit`/`SearchHits`? Can there be an `AggregatedPage` that contains `T` instead of `SearchHit<T>`?

Related (in the context of a new ticket): Would it make sense to rename `AggregatedPage` to `AggregationPage` and let a deprecated `AggregatedPage` extend from `AggregationPage`?

Reply:
`AggregatedPage` extends `ScrolledPage>`and `ScoredPage` (I'm not sure if we need these as separate interfaces, as `AggregatedPageImpl` is the only implementaton of it all, and it adds ES aggregations. I don't like this class as it exposes ES objects in the API, but at the moment it's the only way to return aggregations. That's something I'd like to change later.